examples/ode_models/ODE_Lotka_Volterra_multiple_ways.myst.md

```diff
@@ -0,0 +1,963 @@
+
+```{code-cell} ipython3
+# Make lists with variable for looping
+var_names = [str(s).split("_")[0] for s in list(model.values_to_rvs.keys())[:-1]]
+# Make lists with model results and model names for plotting
+inference_results = [
+    trace_slice,
+    trace_DEMZ,
+    trace_DEM,
+    trace_M,
+    trace_SMC_like,
+    trace_SMC_e1,
+    trace_SMC_e10,
+    trace_pymc_ode,
+    trace_scan,
+]
+model_names = [
+    "Slice Sampler",
+    "DEMetropolisZ",
+    "DEMetropolis",
+    "Metropolis",
+    "SMC with Likelihood",
+    "SMC e=1",
+    "SMC e=10",
+    "PyMC ODE NUTs",
+    "Pytensor Scan NUTs",
+]
+
+# Loop through variable names
+for var_name in var_names:
+    axes = az.plot_forest(
+        inference_results,
+        model_names=model_names,
+        var_names=var_name,
+        kind="forestplot",
+        legend=False,
+        combined=True,
+        figsize=(7, 3),
+    )
+    axes[0].set_title(f"Marginal Probability: {var_name}")
+    # Clean up ytick labels
+    ylabels = axes[0].get_yticklabels()
+    new_ylabels = []
+    for label in ylabels:
+        txt = label.get_text()
+        txt = txt.replace(": " + var_name, "")
+        label.set_text(txt)
+        new_ylabels.append(label)
+    axes[0].set_yticklabels(new_ylabels)
+
+    plt.show();
+```
+
+**Notes:**  
+If we ran the samplers for long enough to get good inferences, we would expect them to converge on the same posterior probability distributions, with the exception of SMC when using a distance parameter (SMC e=1 and SMC e=10).  When using a distance parameter, the model does not include a likelihood function and therefore represents a different posterior probability distribution than the other models.
+
++++
+
+### Key Conclusions
+We performed Bayesian inference on a system of ODEs in 4 main ways: 
+* Scipy `odeint` wrapped in a Pytensor `op` and sampled with non-gradient-based samplers (comparing 5 different samplers).  
+* Scipy `odeint` wrapped in a `pm.Simulator` function and sampled with a non-likelihood-based sequential Monte Carlo (SMC) sampler.  
+* PyMC `ode.DifferentialEquation` sampled with NUTs.  
+* Forward simulation using `pytensor.scan` and sampled with NUTs.  
+
+The "winner" for this problem was the Scipy `odeint` solver with a differential evolution (DE) Metropolis sampler.  The improved efficiency of the NUTs sampler did not make up for the inefficiency in using the slow ODE solvers with gradients.  Sticking with Scipy and DEMetropolis is also the simplest workflow for a scientist with a working numeric model and the desire to perform Bayesian inference.  Just wrapping the numeric model in a Pytensor op and plugging it into a PyMC model can get you a long way!  
+
++++
+
+## Authors
+Organized and rewritten by Greg Brunkhorst from multiple PyMC.io example notebooks by Sanmitra Ghosh, Demetri Pananos, and the PyMC Team.
+
```

Comment:
There should be a "Watermark" heading right before this code cell: https://docs.pymc.io/en/latest/contributing/jupyter_style.html#watermark

Reply:
added heading